tests/queue/queue_shortcuts_usm.h

```diff
@@ -0,0 +1,291 @@
+    T* d_src = sycl::malloc_device<T>(element_count, queue);
+    T* d_dest = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_src, h_src.get(), element_count * sizeof(T));
+      cgh.memcpy(d_dest, h_dest.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the copy from d_src to d_dest using shortcuts
+    sycl::event event = helper(d_dest, d_src);
+    event.wait();
+
+    // copy destination buffer back to host without shortcut functions
+    std::unique_ptr<T[]> h_actual = std::make_unique<T[]>(element_count);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(h_actual.get(), d_dest, element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // check the result
+    for (unsigned int i = 0; i < element_count; i++) {
+      CHECK(((t_test + i) == h_actual[i]));
+    }
+
+    sycl::free(d_dest, queue);
+    sycl::free(d_src, queue);
+
+    return event;
+  }
+
+  template <typename kernel_helper>
+  sycl::event run_memset(kernel_helper helper) {
+    constexpr char init = 0;
+    constexpr char test = 1;
+
+    // set up initial values on host
+    std::unique_ptr<T[]> h_ptr = std::make_unique<T[]>(element_count);
+    std::memset(reinterpret_cast<void*>(h_ptr.get()), init,
+                element_count * sizeof(T));
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_ptr = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_ptr, h_ptr.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the memset from d_src to d_dest using shortcuts
+    sycl::event event = helper(d_ptr, test);
+    event.wait();
+
+    // copy destination buffer back to host without shortcut functions
+    std::unique_ptr<T[]> h_actual = std::make_unique<T[]>(element_count);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(h_actual.get(), d_ptr, element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // check the result
+    char* actual_char = reinterpret_cast<char*>(h_actual.get());
+    for (unsigned int i = 0; i < element_count * sizeof(T); i++) {
+      CHECK((test == actual_char[i]));
+    }
+
+    sycl::free(d_ptr, queue);
+
+    return event;
+  }
+
+  template <typename kernel_helper>
+  sycl::event run_fill(kernel_helper helper) {
+    // set up initial values on host
+    const T t_init{0};
+    const T t_test{1};
+    std::unique_ptr<T[]> h_ptr = std::make_unique<T[]>(element_count);
+    std::fill(h_ptr.get(), h_ptr.get() + element_count, t_init);
+
+    // set initial values of device-allocated usm buffer without using shortcuts
+    T* d_ptr = sycl::malloc_device<T>(element_count, queue);
+    queue.submit([&](sycl::handler& cgh) {
+      cgh.memcpy(d_ptr, h_ptr.get(), element_count * sizeof(T));
+    });
+    queue.wait();
+
+    // perform the memset from d_src to d_dest using shortcuts
```

Comment:
Please, update comments in this function. They seem to be copied from `run_memset` function.

Reply:
I updated the comments.